src/service.js

```diff
@@ -586,9 +588,9 @@ function getServicePromiseOrNullInternal(holder, id) {
       return s.promise;
     } else if (s.obj) {
       return s.promise = Promise.resolve(s.obj);
-    } else {
-      dev().assert((s.build),
-          'Expected object, promise, or builder to be present');
+    } else if (s.build) {
+      s.obj = s.build();
```

Comment:
Let's leave `} else {` and `dev.assert()`. It's still better to assert this in dev since we expect either `promise` or `obj` or `build` to exist.